ctrl/instances/DTD.py

```diff
@@ -25,6 +26,20 @@ def center_crop(img, size):
 
     return img.crop((left, top, right, bottom))
 
+def untar(path):
+    directory_path = os.path.dirname(path)
+    with tarfile.open(path) as tar_file:
+        tar_file.extractall(directory_path)
+
+def download(url, path):
+    if not os.path.exists(path):
+        os.makedirs(path)
+    file_name = os.path.join(path, url.split("/")[-1])
+    if os.path.exists(file_name):
+        print(f"Dataset already downloaded at {file_name}.")
+    else:
+        urllib.request.urlretrieve(url, file_name)
```

Comment:
Could you add the `reporthook` param to log progression since the downloaded file is rather large? 
For exemple you could plug the `gen_bar_updater` from torch.datasets.utils.